src/foam/u2/view/ReadReferenceView.js

```diff
@@ -79,21 +174,73 @@ foam.CLASS({
 
     function fromProperty(prop) {
       this.SUPER(prop);
+      
       this.prop = prop;
-      var dao = this.ctrl.__subContext__[prop.targetDAOKey];
-      this.permissionEnableLinkCheck().then(() => {
-        if ( dao )
+      
+      // fetch link config properties from where they were provided
+      // (i.e., from reference property or this view)
+      this.enableLink = this.prop.enableLink && this.enableLink;
+      this.controlAccessToDAOSummary = this.prop.controlAccessToDAOSummary || this.controlAccessToDAOSummary;
+      this.menuKeys = this.prop.menuKeys.length > 0 ? this.prop.menuKeys : this.menuKeys;
+
+      this.configLink().then(() => {
+        const dao = this.ctrl.__subContext__[prop.targetDAOKey];
+        if ( dao ) {
           dao.find(this.data).then((o) => this.obj = o);
+        }
       });
     },
 
-    async function permissionEnableLinkCheck() {
-      if ( ! this.auth ) return;
-      let permission = `${this.prop.of.id}.${this.prop.name}.disableRefLink`;
-      permission = permission.toLowerCase();
-      await this.auth.check(this.__subContext__, permission).then( check => {
-        this.enableLink = ! check;
-      })
-    }
+    async function configLink() {
+      /*
+       * Uses the flow chart above to control access to the link
+       * and what is set to this link.
+       */
+
+      // enableLink explicitly set to false?
+      if ( ! this.auth || ! this.enableLink ) {
+        this.enableLink = false;
+        this.linkTo = '';
+        return;
+      }      
+      
+      try {
+        // menus are provided?
+        if ( this.menuKeys.length > 0 ) {
+          // check permissions for menus
+          const permissions = await Promise.all([...this.menuKeys].map(menuId => {
+            return this.auth.check(this.__subContext__, `menu.read.${menuId}`);
+          }));
+
+          const firstAt = permissions.indexOf(true);
+          // can read menu?
+          if ( firstAt > -1 ) {
+            this.enableLink = true;
+            this.linkTo = this.menuKeys[firstAt];
+          } else {
+            this.enableLink = false;
+            this.linkTo = '';
+          }
+        // menus not provided
+        } else {
+          // access to dao summary?
+          if ( 
+            ! this.controlAccessToDAOSummary ||
+            await this.auth.check(this.__subContext__, `${this.prop.targetDAOkey}Summary.read`)
```

Comment:
I would rather get rid of 'controlAccessToDAOSummary' entirely, and instead check for the service.targetDAOkey permission, which users will have anyway.

Reply:
This looks like a bug, because isn't the property called targetDAOKey, with a capital "K"?

Reply:
So I don't know about this - if we go with service permissions then lets consider:
service.natureCodeDAO
service.groupDAO

both are given to the treviso-fraud-ops agent - because of approvalRequest look up and line item display on txn ... but we don't want the link enabled for references to these objects (group and natureCode)

Reply:
BUT - we could setup the menuKey on the reference property of concern and not have conflicting issues. I do like service."targetDAOKey" ... so ya I'm good with this.